### DISCUSSION TITLE
Is there a reason why gevent mode uses gevent-websocket instead of websocket-client?

### DISCUSSION THREAD

**Discussion by PawelTroka** [miguelgrinberg/Flask-SocketIO#1785] [Q&A]
Hi!

I am wondering, since gevent-websocket (https://pypi.org/project/gevent-websocket/ https://gitlab.com/noppo/gevent-websocket) seems a bit stale (last update Mar 12, 2017), is there a reason it must be used instead of websocket-client (https://pypi.org/project/websocket-client/ https://github.com/websocket-client/websocket-client) ? Is the former better when used with gevent than the latter? Is monkey patching websocket-client not enough or maybe there are some other reasons for this choice?

**Comment by miguelgrinberg** [comment]
You are confusing a websocket client with a websocket server. The gevent-websocket project is a server, while websocket-client is not.

**Reply by PawelTroka** [reply]
> You are confusing a websocket client with a websocket server. The gevent-websocket project is a server, while websocket-client is not.

Thanks for the reply! That makes total sense. Do you think it is safe to use these two together where one is native gevent library (gevent-websocket) and the other one is using one thread for receiving data (but that of course is monkey patched, so it's not really a thread but a single greenlet)?

**Reply by miguelgrinberg** [reply]
Yeah, my guess is that this isn't going to cause any problems.
